pkg/workloads/reservation.go

```diff
@@ -1360,41 +1365,65 @@ func (a *API) handleKubernetesSize(ctx context.Context, db *mongo.Database, work
 
 func (a *API) handleKubernetesPublicIP(ctx context.Context, db *mongo.Database, workload types.WorkloaderType, userID int64) mw.Response {
 	k8sWorkload := workload.Workloader.(*generated.K8S)
+	return checkPublicIPAvailablity(ctx, db, k8sWorkload.PublicIP, userID)
+}
+
+func (a *API) handleVMPublicIP(ctx context.Context, db *mongo.Database, workload types.WorkloaderType, userID int64) mw.Response {
+	vmWorkload := workload.Workloader.(*generated.VirtualMachine)
+	return checkPublicIPAvailablity(ctx, db, vmWorkload.PublicIP, userID)
+}
+
+func checkPublicIPAvailablity(ctx context.Context, db *mongo.Database, publicIP schema.ID, userID int64) mw.Response {
 
-	if k8sWorkload.PublicIP == 0 {
+	if publicIP == 0 {
 		return nil
 	}
 
 	var err error
 
 	var workloadFiler types.WorkloadFilter
 	workloadFiler = workloadFiler.
-		WithID(k8sWorkload.PublicIP).
+		WithID(publicIP).
 		WithNextAction(generated.NextActionDeploy).
 		WithCustomerID(userID)
 
 	_, err = workloadFiler.Get(ctx, db)
 	if err != nil {
-		return mw.NotFound(errors.Wrapf(err, "ip workload '%d' not found", k8sWorkload.PublicIP))
+		return mw.NotFound(errors.Wrapf(err, "ip workload '%d' not found", publicIP))
 	}
-
 	// Check if there is already a k8s workload with this public ip reservation in the database
-	workloadFiler = workloadFiler.
+	workloadFiler = types.WorkloadFilter{}.
 		WithCustomerID(userID).
 		WithNextAction(generated.NextActionDeploy).
 		WithWorkloadType(generated.WorkloadTypeKubernetes).
-		WithPublicIP(k8sWorkload.PublicIP)
+		WithPublicIP(publicIP)
 
+	// to be tested on a node with pubip
 	_, err = workloadFiler.Get(ctx, db)
-	if err == mongo.ErrNoDocuments {
-		// If there is no match, this means this is a valid reservation
-		return nil
-	} else if err != nil {
+	if err == nil {
+		// some documents are returened -> ip in use
+		return mw.Conflict(fmt.Errorf("public ip is in use"))
+	} else if err != nil && err != mongo.ErrNoDocuments {
+		// some error occured other than no documents found
+		return mw.Error(err)
+	}
+
+	// Check if there is already a vm workload with this public ip reservation in the database
+	workloadFiler = workloadFiler.
```

Comment:
This is wrong, a filter object is not reusable, because now this just "Adds" to the criteria that was added before at line 1395. You need to create a new filter same way as done on 1395

Reply:
Also i really think you can get all the results u want just by dropping the type filter at line 1398. this way u will find if the public_ip is set on any active reservation.

NOTE: for optimizations the `public_ip` key is probably needed to be added to the indexes list in `pkg/workloads/types/setup.go` for the `WorkloadCollection` collection